airbyte-integrations/connectors/source-freshdesk/source_freshdesk/api.py

```diff
@@ -266,6 +266,65 @@ def list(self, fields: Sequence[str] = None) -> Iterator[dict]:
         params = {"include": "description"}
         yield from self.read(partial(self._api_get, url="tickets"), params=params)
 
+    # This block extends TicketsAPI Stream to overcome '300 page' server error.
+    # Since the TicketsAPI Stream list has a 300 page pagination limit, after 300 pages, update the parameters with
+    # query using 'updated_since' = last_record, if there is more data remaining.
+    def get_tickets(self, getter: Callable, params: Mapping[str, Any] = None) -> Iterator:
+        """Read using getter"""
+        params = params or {}
+        # the maximum page allowed to pull during pagination.
+        ticket_paginate_limit = 300
+
+        # Reading Tickets up to the `ticket_paginate_limit`
+        for page in range(1, ticket_paginate_limit):
+            batch = list(getter(params={**params, "per_page": self.result_return_limit, "page": page}))
+            yield from batch
+
+            if len(batch) < self.result_return_limit:
+                return iter(())
+            else:
+                # position 0, because the records are returned in 'desc'
+                last_record = batch[0]["updated_at"]
+
+        # if there is more after 300 pages
+        # Define last_record = `updated_at` + 1 sec
+        # query using 'updated_since' = last_record + pagination
+        last_record = pendulum.parse(last_record).add(seconds=1)
+        for more_page in range(1, self.maximum_page):
```

Comment:
this block should be inside a while loop. What if there are more than 600 pages? 

Also if possible should be DRY'd with the one above it

Reply:
Changed the for to while loop, now we would fetch till the data exists in the tickets stream, no matter how many pages there are. Each 300th iteration the request parameters would make a checkpoint on the latest fetch and start the new circle from that, basically, we paginate the pagination every 300th pages.